src/diffusers/quantizers/quantization_config.py

```diff
@@ -389,3 +392,272 @@ def to_diff_dict(self) -> Dict[str, Any]:
                 serializable_config_dict[key] = value
 
         return serializable_config_dict
+
+
+@dataclass
+class TorchAoConfig(QuantizationConfigMixin):
```

Comment:
high level I feel we also want to add an API to quantize the full model, this will be important if we want to use autoquant, which works on a full model instead of modules

Reply:
also wondering if we can share this with transformers library as well, since the integration is mostly the same

Reply:
Yes, we'll have to plan how to support that. For the moment, we're going to try and merge this in after addressing the reviews and another round of testing. In a follow up PR, we can do the autoquant integration

Reply:
> high level I feel we also want to add an API to quantize the full model, this will be important if we want to use autoquant, which works on a full model instead of modules

Yes, we plan to do that in a follow-up but since quantizing with `autoquant` is just a one-liner, anyway, I think we could suggest that from our docs. 

> also wondering if we can share this with transformers library as well, since the integration is mostly the same

This means users will have to always install `transformers` and we cannot control anything that might need changing specific to `diffusers`. So, we prefer to keep a (modified) copy of our own in the repo. We followed the same for `bitsandbytes`.

Reply:
integrating autoquant into `TorchAoConfig` will have the benefit of adding it to optimum: https://github.com/huggingface/optimum-benchmark/pull/297 and leaderboard: https://huggingface.co/spaces/optimum/llm-perf-leaderboard in the future

Reply:
> This means users will have to always install transformers and we cannot control anything that might need changing specific to diffusers. So, we prefer to keep a (modified) copy of our own in the repo. We followed the same for bitsandbytes.

oh makes sense to separate for now, I'm also not suggesting to depend on transformers for this, more of putting the integration else where that can be used by both. maybe in torchao (not sure if it's possible)